apps/src/lab2/responseValidators.ts

```diff
@@ -1,32 +1,49 @@
 import {ResponseValidator} from '@cdo/apps/util/HttpClient';
-import {BlocklySource, LevelProperties, ProjectSources} from './types';
+import {AppName, BlocklySource, LevelProperties, ProjectSources} from './types';
+import Lab2Registry from './Lab2Registry';
+import {BLOCKLY_LABS} from './constants';
+import {NestedSourceCode} from '@cdo/apps/pythonlab/pythonlabRedux';
 
-export const SourceResponseValidator: ResponseValidator<
```

Comment:
Potentially overkill - but just as a thought, what do you think of keeping a top-level `SourceResponseValidator` that internally delegates to the appropriate validator for the current app name? That way instead of storing a validator object on the registry, we can just store the current app name, and this top-level validator could look up the app name in the registry and pick the right delegate internally. And then the sourcesApi could still use the top-level validator without needing to do a lookup in the registry.

Reply:
that's a good idea! I was stuck on how we get the app name but storing it on the registry makes sense, and it's likely some other feature will need that info in the future anyway.

Reply:
This was a nice refactor suggestion, thanks, @sanchitmalhotra126!